pulpcore/pulpcore/app/viewsets/repository.py

```diff
@@ -178,6 +187,63 @@ def destroy(self, request, repository_pk, number):
         )
         return OperationPostponedResponse([async_result], request)
 
+    def validate_content_url(self, url):
+        """
+        Validates that the path belongs to a Content resource
+
+        Args:
+            url (str): href that needs validation
+
+        Returns:
+            PK of the Content
+
+        Raises:
+            :class:`rest_framework.exceptions.ValidationError`: When the URL is not a valid
+                Content resource
+        """
+
+        try:
+            resolver = resolve(urlparse(url).path)
+        except Resolver404 as e:
+            raise ValidationError(detail=_("Invalid hyperlink. %s") % url)
+
+        if not resolver.view_name.startswith('content') or not resolver.view_name.endswith(
+                'detail'):
+            # URL exists, but does not correspond to Content resource
+            raise ValidationError(detail=_("Invalid hyperlink. %s") % url)
+
+        content_field = serializers.HyperlinkedRelatedField(view_name=resolver.view_name,
+                                                            queryset=Content.objects.all(),
+                                                            source='*', initial=url)
+        try:
+            content_field.run_validation(data=url)
+        except ValidationError as e:
+            # Append the URL of missing Content to the error message
+            e.detail[0] = "%s %s" % (e.detail[0], url)
+            raise e
+        return resolver.kwargs['pk']
+
+    def create(self, request, repository_pk):
+        add_content_units = []
+        remove_content_units = []
+        if 'add_content_units' in request.data:
+            for url in request.data['add_content_units']:
+                add_content_units.append(self.validate_content_url(url))
```

Comment:
A problem with this pattern is that the content_urls are validated synchronously. It is conceivable that a user issues a sync task and then a manual remove. They might know the sync will add some packages that they want to remove, so they might do another task to create a new repository version that removes them. This will fail synchronously because the units don't exist yet. 

I wonder if it would be better to just validate the form of the hrefs, to make sure that they could be valid. Then in the task, we just try it and inform the users if any hrefs were not added because they didn't exist.